### PR TITLE
Feat/volume replication

### PR DESCRIPTION
**Pull request by MiroslavRepka**
This PR fixes the issue described in #746 by implementing more checks and cordoning the nodes before Claudie will move Replicas of Volumes. This was tested by migrating the ETDC cluster to multiple new node pools, where each migration resulted in the same state of the database.

There might occasionally be still one issue, which is not severe, as all replicas will migrate with the data correctly. The issue is described as a note in [one comment](https://github.com/berops/claudie/compare/feat/volume-replication?expand=1#diff-83286613a1db2b0823d24bedfdef1758c39d2ac7f08a39a1a8b491fc0930d0bbR111-R116), with the link to the discussion I started with Longhorn maintainers.

### CODE REVIEW

File: services/kuber/server/nodes/delete.go [hunk 1]
```diff
@@ -79,22 +81,42 @@ func (d *Deleter) DeleteNodes() (*pb.K8Scluster, error) {
...
+		// Delete worker nodes
 		if err := d.deleteNodesByName(kubectl, worker, realNodeNames); err != nil {
 			return nil, fmt.Errorf("error while deleting nodes from worker nodes for %s : %w", d.clusterPrefix, err)
 		}
+		// NOTE: Might need to verify if the volume got detached. The issue is still being discussed in #longhorn channel in CNCF slack workspace.
+		// https://cloud-native.slack.com/archives/CNVPEL9U3/p1683744899327089
+		// Issue seen, when once the node is deleted, some volumes might be still attached to the already deleted node, even though all replicas
+		// have moved to other nodes in cluster. Once manually detaching the volume from already deleted node (via UI), it attached again to a new node
+		// with no issues. Since there is no easy way how to verify and detach volume in some CLI way (due to longhorn policy of  doing everything via UI
+		// in ClickOps), leaving this note until discussion with longhorn maintainers concludes.
```
Comment (Despire) [inline services/kuber/server/nodes/delete.go:116]: Could we open a new Issue for this so we don't forget about this ?

**Review by Despire — APPROVED** [review]
@MiroslavRepka LGTM 👍

File: services/kuber/server/nodes/delete.go (hunk 1 above)
Comment (MiroslavRepka) [reply services/kuber/server/nodes/delete.go:116]: They just asked me to open an issue for it in the Longhorn repository. Should we open an issue on our end as well?

Comment (Despire) [reply services/kuber/server/nodes/delete.go:116]: Yes, and Link it with the Longhorn Issue

Comment (MiroslavRepka) [reply services/kuber/server/nodes/delete.go:116]: https://github.com/berops/claudie/issues/784

**Pull request merged**